src/pool-bin/BinPoolManager.sol

```diff
@@ -329,11 +317,7 @@ contract BinPoolManager is IBinPoolManager, ProtocolFees, Extsload {
 
         IBinHooks hooks = IBinHooks(address(key.hooks));
         if (key.parameters.shouldCall(HOOKS_BEFORE_DONATE_OFFSET, hooks)) {
-            bytes4 selector = hooks.beforeDonate(msg.sender, key, amount0, amount1, hookData);
-            if (key.parameters.isValidNoOpCall(HOOKS_NO_OP_OFFSET, selector)) {
-                // Sentinel return value used to signify that a NoOp occurred.
-                return (BalanceDeltaLibrary.MAXIMUM_DELTA, binId);
-            } else if (selector != IBinHooks.beforeDonate.selector) {
+            if (hooks.beforeDonate(msg.sender, key, amount0, amount1, hookData) != IBinHooks.beforeDonate.selector) {
```

Comment:
im thinking if we can create an issue to refactor (not in this PR but after all parts are merged) -- eg. move all hook call to CLHooks to make it easier to read for the devs. what do you think? 

```
// without CLHooks
ICLHooks hooks = ICLHooks(address(key.hooks));
        if (key.parameters.shouldCall(HOOKS_BEFORE_DONATE_OFFSET, hooks)) {
            if (hooks.beforeDonate(msg.sender, key, amount0, amount1, hookData) != ICLHooks.beforeDonate.selector) {
                revert Hooks.InvalidHookResponse();
            }
        }

// using CLHooks
(delta, hookDelta) = CLHooks.afterModifyLiquidity(key, params, delta + feeDelta, hookData);
```

Reply:
Agree with that, it brings better readability